src/world.cpp

```diff
@@ -2208,7 +2208,7 @@ void World_GenRoom(struct room_s *room, class VT_Level *tr)
     room->alternate_room = NULL;
     room->base_room = NULL;
 
-    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count))
+    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count) && (room->id < tr_room->alternate_room))
```

Comment:
May be I didn't understand it correctly, but using the code:
```
#include <stdio.h>

void swap(int *a, int *b) {
	int t;
	t = *a;
	*a = *b;
	*b = t;
	return;
}

int main() {
  int rooms[] = {1,2,3,4}, room1, room2;
  for (room1 = 0; room1 < sizeof(rooms)/sizeof(int)-1; room1++) {
	  if (rooms[room1] < 0) continue;
	  room2 = rooms[room1];
	  swap(&rooms[room1], &rooms[room2]);
	  rooms[room2] = -1;
  };
  printf("[%d,%d,%d,%d]\n",rooms[0],rooms[1],rooms[2],rooms[3]);
  return 0;
}
```
I got `[2,-1,4,-1]`.

I found how the engine determined a starting room. It just used PORTAL function. It also used initial Lara position without transferring her into another room.